llvm/lib/SYCLLowerIR/LowerESIMD.cpp

```diff
@@ -626,6 +628,9 @@ static void translateSLMInit(CallInst &CI) {
 static void translatePackMask(CallInst &CI) {
   using Demangler = id::ManglingParser<SimpleAllocator>;
   Function *F = CI.getCalledFunction();
+  if (!F)
+    return;
```

Comment:
I suggest replacing `getCalledFunction` result checks with `assert` to avoid unnecessary overhead at compile time.

Reply:
the same can be done for getNamedMetadata either

Reply:
I thought Klockwork would still complain about asserts, but it looks like it's not the case.